src/main/java/ru/test/restservice/service/CompilerService.java

```diff
@@ -0,0 +1,83 @@
+package ru.test.restservice.service;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static ru.test.restservice.MainApplication.isWindows;
+
+public class CompilerService {
+
+    // TODO: перенести все эксепшны в хандлер
+    @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
+    public static class CompilationException extends RuntimeException {
+        public CompilationException(String message) {
+            super(message);
+        }
+    }
+
+    // TODO: добавить функцию сохранения изменений в файле
+
+    /**
+     * Метод, создающий в папке пользователя новый .tex файл
+     * или перезаписывающий уже существующий
+     *
+     * @param text содержимое создаваемого .tex файла
+     */
+    public static void createTexFile(String text) {
+        Path path = Paths.get("test/test.tex");
+        try {
+            if (!Files.exists(Paths.get("test"))) {
+                Files.createDirectory(Paths.get("test"));
+            }
```

Comment:
Можно убрать проверку
```suggestion
            Files.createDirectories(Paths.get("test"));
```